datatypes/src/raster/geo_transform.rs

```diff
@@ -175,6 +154,61 @@ impl GeoTransform {
         }
     }
 
+    /// compute the index of the upper left pixel that is contained in the `partition`
+    pub fn upper_left_pixel_idx(&self, partition: &SpatialPartition2D) -> GridIdx2D {
+        // choose the epsilon relative to the pixel size
+        const EPSILON: f64 = 0.000_001;
+        let epsilon: Coordinate2D =
+            (self.x_pixel_size() * EPSILON, self.y_pixel_size() * EPSILON).into();
+
+        let upper_left_coordinate = partition.upper_left() + epsilon;
```

Comment:
Why should we do this?
I this is reasonable, please add a comment.

Reply:
see comment above